### DISCUSSION TITLE
VPN working, but queing QUICK_MODE

### DISCUSSION THREAD

**Discussion by DariusSki** [strongswan/strongswan#2290] [General]
Hi all,

we have been a racoon user on earlier Debians, and moved to strongswan on Debian12.
Out of several IPSec VPNs, two are having a same problem, they come up, traffic is routed, but QUICK_MODE tasks are being queued:

     bite109[1847]: ESTABLISHED 2 minutes ago, ed_our_ip[ed_our_ip]...ed_their_ip[ed_their_ip]
     bite109[1847]: IKEv1 SPIs: aa30d2936831dc28_i 27f76a4906404bf3_r*, pre-shared key reauthentication in 23 hours
     bite109[1847]: IKE proposal: 3DES_CBC/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
     bite109[1847]: Tasks queued: QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE
     bite109[1847]: Tasks active: QUICK_MODE
     bite109{378}:  INSTALLED, TUNNEL, reqid 1, ESP SPIs: c8ca0b93_i 5496e679_o
     bite109{378}:  3DES_CBC/HMAC_MD5_96, 32388105 bytes_i (43291 pkts, 0s ago), 1930582 bytes_o (29279 pkts, 0s ago), rekeying in 23 hours
     bite109{378}:   10.111.111.1/32 === 10.109.0.0/16

It is looking like this, sometimes one or both will work completely fine after restart, but eventually ( maybe after IKE reauth every 24 hours? will start queing QUICK_MODE tasks.

Also a side effect of this problem, seems to be 
"  bite109[1847]: ESTABLISHED 2 minutes ago, " => it is reestablished every 2 minutes?

Config looks like this in /etc/ipsec.conf 

conn bite109
        type=tunnel
        auto=route
        keyexchange=ikev1
        authby=secret
        left=ed_our_ip
        leftsubnet=10.111.111.1/32
        right=ed_their_ip
        rightsubnet=10.109.0.0/16
        ike=3des-sha1-modp1024!
        compress=yes
        esp=3des-md5
        aggressive=no
        keyingtries=%forever
        ikelifetime=86400s
        lifetime=86400s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=none
//      modeconfig=push
//  we have tried modeconfig=push as seen in some random discussions about strongswan, but it did not help it seems

Both VPN's are from like 2017 era, so probably some sort of Cisco or ASA boxen, but we don't really have any visibility into what's on other side.

I'd be very thankful about any suggestions.

Darius.

**Comment by tobiasbrunner** [comment]
Yeah, without complete log it's difficult to say what's happening (see #196 for recommended settings).

> Also a side effect of this problem, seems to be
> " bite109[1847]: ESTABLISHED 2 minutes ago, " => it is reestablished every 2 minutes?

Why would you read it like that?

**Reply by DariusSki** [reply]
> Why would you read it like that?

Because it never goes above 2 minutes when QUICK_MODE's are queued and goes to rekey interval when it sometimes starts correctly, sorry I wasn't clear enough.

For example right now:

Connections:
     bite109:  ip...ip  IKEv1
     bite109:   local:  [ip] uses pre-shared key authentication
     bite109:   remote: [ip] uses pre-shared key authentication
     bite109:   child:  10.111.111.1/32 === 10.109.0.0/16 TUNNEL
Routed Connections:
     bite109{2}:  ROUTED, TUNNEL, reqid 1
     bite109{2}:   10.111.111.1/32 === 10.109.0.0/16
Security Associations (5 up, 1 connecting):
     bite109[12088]: ESTABLISHED 10 seconds ago, ip[ip]...ip[ip]
     bite109[12088]: IKEv1 SPIs: b437dc85616c5911_i f92700b415fce9fb_r*, pre-shared key reauthentication in 23 hours
     bite109[12088]: IKE proposal: 3DES_CBC/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
     bite109[12088]: Tasks queued: QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE QUICK_MODE (many more)
     bite109[12088]: Tasks active: QUICK_MODE
     bite109{2462}:  INSTALLED, TUNNEL, reqid 1, ESP SPIs: c8a5ce4b_i f3454e83_o
     bite109{2462}:  3DES_CBC/HMAC_MD5_96, 831539 bytes_i (2686 pkts, 0s ago), 421424 bytes_o (2857 pkts, 0s ago), rekeying in 23 hours
     bite109{2462}:   10.111.111.1/32 === 10.109.0.0/16


I have read the guidelines before posting, but opted to skip the log section due to the box serving multiple IPSec tunnels and logs being super chatty due to that.

Also there is a problem on debian 12, that configuration of logging and strongswan overall is super mystical.
We are currently just editing /etc/ipsec.conf and controlling start up with /etc/init.d/ipsec start and friends.

By default it is pumping logs to journalctl and with some playing with variuos config files i was only able to arrive to log "control" via;

/etc/ipsec.conf
# ipsec.conf - strongSwan IPsec configuration file

# basic configuration

config setup
        charondebug = dmn -1, mgr -1, ike 1 , chd -1, job -1, cfg -1, knl -1, net -1, asn -1, enc -1, lib -1, esp -1, tls -1, tnc -1, imc -1, imv -1, pts -1


Am i correct to assume that 
charondebug = dmn -1, mgr -1, **ike 1** , chd -1, job -1, cfg -1, knl -1, **net 1**, **asn 1**, **enc 1**, lib -1, esp -1, tls -1, tnc -1, imc -1, imv -1, pts -1

is something we need for proper logs that would help to debug the issue( in journalctl )?

Thanks!

**Reply by tobiasbrunner** [reply]
> Because it never goes above 2 minutes when QUICK_MODE's are queued and goes to rekey interval when it sometimes starts correctly, sorry I wasn't clear enough.

Hm, but Quick Mode exchanges have nothing to do with the IKE_SA. So there must be something else going on here (the log might tell us).

> Also there is a problem on debian 12, that configuration of logging and strongswan overall is super mystical.

What do you mean?

> Am i correct to assume that charondebug = dmn -1, mgr -1, **ike 1** , chd -1, job -1, cfg -1, knl -1, **net 1**, **asn 1**, **enc 1**, lib -1, esp -1, tls -1, tnc -1, imc -1, imv -1, pts -1
> 
> is something we need for proper logs that would help to debug the issue( in journalctl )?

Use the levels given in #196 (if you use `charondebug`, which I wouldn't recommend, you have to manually set everything not set to 1 to 2 as there is no `default`).

**Reply by DariusSki** [reply]
> What do you mean?

We had problems configuring logging (wanted to reduce spam to systemd journal), and ended up editing /etc/ipsec.conf
So with default install:

apt-get install strongswan

resulting in 
ii  libstrongswan                     5.9.8-5+deb12u1                  amd64        strongSwan utility and crypto library
ii  libstrongswan-standard-plugins    5.9.8-5+deb12u1                  amd64        strongSwan utility and crypto library (standard plugins)
ii  strongswan                        5.9.8-5+deb12u1                  all          IPsec VPN solution metapackage
ii  strongswan-charon                 5.9.8-5+deb12u1                  amd64        strongSwan Internet Key Exchange daemon
ii  strongswan-libcharon              5.9.8-5+deb12u1                  amd64        strongSwan charon library
ii  strongswan-starter                5.9.8-5+deb12u1                  amd64        strongSwan daemon starter and configuration file parser


There are two ways/places to edit things => /etc/ipsec.conf and /etc/strongswan.conf+/etc/strongswan.d/ files that seem to control some of the same things and interact in ways that are not obviuos. We might have made some trivial mistake in strongswan logging config, but it did not work with some effort and /etc/ipsec.conf started working immediately.

( as mentioned in first letter, we have just switched to strongswan, and used racoon / setkey in the past on Debian 9  so if above is stupid, just ignore it :) )

**Reply by DariusSki** [reply]
[bite109.log](https://github.com/user-attachments/files/15864865/bite109.log)

file logging seems to worked just fine, please find the file until first QUICK_MODE's are queued:

for someone with debian12 also reading this, edit /etc/strongswan.d/charon-logging.conf and the filelog should look like this )
charon {
    filelog {
        charon-debug-log {
                path = /var/log/charon_debug.log
                time_format = %a, %Y-%m-%d, %H:%M:%S
                default = 2
                net = 1
                enc = 1
                asn = 1
                job = 1
                ike_name = yes
                append = no
                flush_line = yes
        }
    }
)

**Comment by DariusSki** [comment]
I have attached log file with ~1 hour of action, lots of QUICK_MODE queing in the file, please let me know if additional log information is needed, seems we got logging figured out now!
[bite109_hour.log](https://github.com/user-attachments/files/15865900/bite109_hour.log)

**Reply by tobiasbrunner** [reply]
We see a lot of the following when your end is initiator of a quick mode request:
```
Mon, 2024-06-17, 12:38:35 02[ENC] <bite109|4> parsed INFORMATIONAL_V1 request 2399364564 [ HASH N((10)) ]
Mon, 2024-06-17, 12:38:35 02[IKE] <bite109|4> received (10) error notify
```
Retransmits are sent for the request and eventually this causes the IKE_SA to get restarted. That error notify is INVALID-PROTOCOL-ID, which is a bit weird.

However, we also see the following when your end is responder to a quick mode request:
```
Mon, 2024-06-17, 12:38:29 02[IKE] <bite109|10> expected IPComp proposal but peer did not send one, IPComp disabled
```
So this indicates to me that your peer can't handle IPComp at all (strongSwan sends a proposal with and one without IPComp, the peer is meant to select the one it supports) and you should probably remove/disable `compress=yes` in your config.

**Reply by DariusSki** [reply]
Thanks, that was the problem. Disabled IPComp in config for boths tunnels are they are stable once more.

I think at some point the other end changed hardware into one not supporting IPComp and racoon/setkey were more forgiving about it.

Also being the responder/initiator might explain why these tunnels were working sometimes without this problem, but would start queing after some time.

Thanks again, awesome community!
